src/renderers/shared/fiber/ReactFiberStack.js

```diff
@@ -34,6 +34,21 @@ exports.createCursor = function<T>(defaultValue : T) : StackCursor<T> {
   };
 };
 
+exports.getPrevious = function<T>(
+  cursor : StackCursor<T>,
+) : T | null {
+  if (index < 0) {
+    if (__DEV__) {
+      warning(false, 'Unexpected read of previous.');
+    }
+    return null;
+  }
+  if (index === 0) {
+    return null;
+  }
+  return valueStack[index - 1];
```

Comment:
Isn't previous actually the value at `valueStack[index]`?

Current is in `cursor.current` and previous is the top item in the stack (at `index`).